lib/report_retryables.ts

```diff
@@ -0,0 +1,439 @@
+import axios from 'axios'
+import { L1ToL2MessageStatus } from '@arbitrum/sdk'
+import { BigNumber, ethers, providers } from 'ethers'
+import { ArbGasInfo__factory } from '@arbitrum/sdk/dist/lib/abi/factories/ArbGasInfo__factory'
+import {
+  ARB_GAS_INFO,
+  ARB_RETRYABLE_TX_ADDRESS,
+} from '@arbitrum/sdk/dist/lib/dataEntities/constants'
+import { ArbRetryableTx__factory } from '@arbitrum/sdk/dist/lib/abi/factories/ArbRetryableTx__factory'
+import { Provider } from '@ethersproject/abstract-provider'
+import { ChildNetwork } from './find_retryables'
+import { slackMessageRetryablesMonitor } from './slack'
+
+export interface ParentChainTicketReport {
+  id: string
+  transactionHash: string
+  sender: string
+  retryableTicketID: string
+}
+
+export interface ChildChainTicketReport {
+  id: string
+  retryTxHash: string
+  createdAtTimestamp: string
+  createdAtBlockNumber: number
+  timeoutTimestamp: string
+  deposit: string
+  status: string
+  retryTo: string
+  retryData: string
+  gasFeeCap: number
+  gasLimit: number
+}
+
+export interface TokenDepositData {
+  l2TicketId: string
+  tokenAmount?: string
+  sender: string
+  l1Token: {
+    symbol: string
+    id: string
+    decimals: number
+  }
+}
+
+export const reportFailedTicket = async ({
+  parentChainTicketReport,
+  childChainTicketReport,
+  tokenDepositData,
+  childChain,
+}: {
+  parentChainTicketReport: ParentChainTicketReport
+  childChainTicketReport: ChildChainTicketReport
+  tokenDepositData?: TokenDepositData
+  childChain: ChildNetwork
+}) => {
+  // slack it
+  const t = childChainTicketReport
+  const now = Math.floor(new Date().getTime() / 1000) // now in s
+
+  // don't report tickets which are not yet scheduled if they have been created in last 2h
+  const reportingPeriodForNotScheduled = 2 * 60 * 60 // 2 hours in s
+  if (
+    t.status == 'NOT_YET_CREATED' &&
+    now - +t.createdAtTimestamp < reportingPeriodForNotScheduled
+  ) {
+    return
+  }
+
+  // don't report tickets which expired more than 2 days ago
+  const reportingPeriodForExpired = 2 * 24 * 60 * 60 // 2 days in s
+  if (
+    t.status == 'EXPIRED' &&
+    now - +t.timeoutTimestamp > reportingPeriodForExpired
+  ) {
+    return
+  }
+
+  const l1Report = parentChainTicketReport
+
+  const childChainProvider = new providers.JsonRpcProvider(
+    String(childChain.orbitRpcUrl)
+  )
+  const parentChainProvider = new providers.JsonRpcProvider(
```

Comment:
Looks like we don't use the `parentChainProvider` anywhere in the script, should we remove it then?

Reply:
Updated 👍